api/main_endpoints/util/userHelpers.js

```diff
@@ -202,6 +202,23 @@ function checkIfPageCountResets(lastLogin) {
 
   return lastLoginWasOverOneWeekAgo || aSundayHasPassedSinceLastLogin;
 }
+// updates users available pages
+async function subtractUserPages(userId, pagesToPrint){
+  const user = await User.findById(userId);
+  if(!user){
+    throw new Error('user not found'); // checks user
+  }
+  if(!Number.isInteger(pagesToPrint) || pagesToPrint <= 0){
+    throw new Error('invalid number');
+  }
+  if (user.pagesPrinted < pagesToPrint) {
+    throw new Error('no pages remaining');
+  }
```

Comment:
instead of throwing an error just do logger.error here and return early, we dont have to return anything